zstacklib/zstacklib/test/lvm/test_safe_write_file.py

```
@@ -0,0 +1,43 @@
+
+        os.remove(dest)
+        with self.assertRaises(Exception) as e:
+            with linux.CrashSafeFileEditor(dest) as content:
+                content.text, _ = re.subn(r"^.*iscsid\.startup.*=.*$", "iscsid.startup = /bin/systemctl start iscsid.socket xxxxxxxxx",
+                                          content.text, flags=re.MULTILINE)
+        print(str(e))
+
+        linux.touch_file(dest)
+        with self.assertRaises(Exception) as ee:
+            with linux.CrashSafeFileEditor(dest) as content:
+                raise Exception("on purpose")
```

Ruff (0.14.14)

Create your own exception

(TRY002)